terraform/040-id-broker/main.tf

```diff
@@ -247,10 +252,20 @@ module "cron_task" {
 /*
  * Create Cloudflare DNS record(s)
  */
+resource "cloudflare_record" "public" {
+  count = var.create_dns_record ? 1 : 0
+
+  zone_id = data.cloudflare_zone.domain.id
+  name    = var.subdomain
+  value   = cloudflare_record.brokerdns.hostname
+  type    = "CNAME"
+  proxied = true
+}
```

Comment:
Are you sure we want a non-region-specific DNS name? If so, we'll need to update our DR / failover processes to include changing which region-specific subdomain it's a CNAME for, and I think we will always want each region's copy of the IDP to always/only call it's own region's instance of ID Broker.

I also added a comment to this PR's YouTrack issue, if you want to address that before / in-tandem-with this question.

Reply:
> Are you sure we want a non-region-specific DNS name? If so, we'll need to update our DR / failover processes to include changing which region-specific subdomain it's a CNAME for, and I think we will always want each region's copy of the IDP to always/only call it's own region's instance of ID Broker.

I may be wrong, but I think the DNS name must not have the region in it so NodePing and the UI can access it. You're right that the failover process will be impacted.

Reply:
Copied from email from @forevermatt:

> On further thought, I believe we probably have (at least most of) the answers to the questions I put on this calendar invitation:
>
>  Do we need a region-less domain name?
>    Yes, for NodePing and profile-ui
>
>  Would the other IDP services use it, or continue using region-specific domain names?
>    Most of them would continue to use the internal, region-specific domain name.
>
>  What about profile-ui: would it need to use the region-less domain name?
>    Since this is a domain name used by end users' browsers, it would need to be the region-less domain name, so our disaster-recovery tools/instructions will just need to address switching that over, too.